src/kv/memstore.rs

```diff
@@ -24,47 +24,49 @@ impl Default for MemStore {
 #[async_trait(?Send)]
 impl Store for MemStore {
     async fn read<'a>(&'a self) -> Result<Box<dyn Read + 'a>> {
-        Ok(Box::new(ReadTransaction::new(self)))
+        let guard = self.map.read().await;
+        Ok(Box::new(ReadTransaction::new(guard)))
     }
 
     async fn write<'a>(&'a self) -> Result<Box<dyn Write + 'a>> {
-        Ok(Box::new(WriteTransaction::new(self)))
+        let guard = self.map.write().await;
+        Ok(Box::new(WriteTransaction::new(guard)))
     }
 }
 
 struct ReadTransaction<'a> {
-    store: &'a MemStore,
+    guard: RwLockReadGuard<'a, HashMap<String, Vec<u8>>>,
```

Comment:
`guard` seems like kind of an odd name for this since it actually contains the data too. Maybe I'd just call it `map`, or `guarded_map` ?